src/k8s-engine/wsl.ts

```diff
@@ -1104,6 +1104,33 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     await this.execCommand('/usr/local/bin/wsl-service', service, 'start');
   }
 
+  /**
+   * Verify that the given command runs successfully
+   * @param command
+   * @protected
+   */
+  protected async verifyReady(...command: string[]) {
+    const startTime = Date.now();
+    const maxWaitTime = 60_000;
+    const waitTime = 200;
+
+    while (true) {
+      const currentTime = Date.now();
+
+      if ((currentTime - startTime) > maxWaitTime) {
+        console.log(`Waited more than ${ maxWaitTime / 1000 } secs for ${ command.join(' ') } to succeed. Giving up.`);
+        break;
+      }
+      try {
+        await this.execCommand(...command);
+        break;
+      } catch (err) {
+        console.log(`Command ${ command } failed: ${ err }:`, err);
+      }
+      await util.promisify(setTimeout)(waitTime);
```

Comment:
Is this essentially just sleeping for 0.2s? If yes, then I think I would increase the sleep to 1s instead.

Reply:
Once I got this working correctly it only needed one retry to succeed for both engines. How about if we do some exponential backoff then, maxing out at 1 sec, because it doesn't seem to need much time to be ready?

Reply:
Too complicated. It takes over a minute to start up the VM, so adding 0.8s doesn't make much of a difference. Maybe change it to 0.5s, or leave it as-is, but don't change the logic itself.

Reply:
Waiting for 1/2 second now.